### PR TITLE
"Fix" some undefined macros on Windows

### PR DESCRIPTION
**Pull request by tniessen**
This addresses https://github.com/cjihrig/uvwasi/issues/11#issuecomment-541383475. Weirdly enough, Windows does not define the `S_IS...` macros, but does define `S_IF...`.

### CODE REVIEW

File: src/fd_table.c [hunk 1]
```diff
@@ -114,14 +116,15 @@ static uvwasi_errno_t uvwasi__get_type_and_rights(uv_file fd,
...
+  if ((mode & S_IFMT) == S_IFREG) {
     *type = UVWASI_FILETYPE_REGULAR_FILE;
     *rights_base = UVWASI__RIGHTS_REGULAR_FILE_BASE;
     *rights_inheriting = UVWASI__RIGHTS_REGULAR_FILE_INHERITING;
-  } else if (S_ISDIR(mode)) {
+  } else if ((mode & S_IFMT) == S_IFDIR) {
     *type = UVWASI_FILETYPE_DIRECTORY;
     *rights_base = UVWASI__RIGHTS_DIRECTORY_BASE;
     *rights_inheriting = UVWASI__RIGHTS_DIRECTORY_INHERITING;
+#ifdef S_ISSOCK
```
Comment (cjihrig) [inline src/fd_table.c]: Does `S_IFSOCK` exist on Windows?

File: src/fd_table.c [hunk 2]
```diff
@@ -134,15 +137,20 @@ static uvwasi_errno_t uvwasi__get_type_and_rights(uv_file fd,
 
     *rights_base = UVWASI__RIGHTS_SOCKET_BASE;
     *rights_inheriting = UVWASI__RIGHTS_SOCKET_INHERITING;
+#endif
+#ifdef S_ISFIFO
```
Comment (cjihrig) [inline src/fd_table.c]: Same question about `S_IFIFO` and `S_IFBLK` on Windows.

File: src/fd_table.c (hunk 1 above)
Comment (tniessen) [reply src/fd_table.c]: No, because Windows does not treat sockets as files.

File: src/fd_table.c (hunk 2 above)
Comment (tniessen) [reply src/fd_table.c]: They don't exist, but I am not 100% sure about the reason. I believe Windows does not treat pipes and block devices as files either, but I might be wrong.

**Comment by cjihrig** [comment]
I landed this, but we might want to revisit and see if `uv_guess_handle()` can make this logic a little better.

**Pull request merged**